client/driver/docker_coordinator.go

```diff
@@ -165,23 +179,44 @@ func (d *dockerCoordinator) PullImage(image string, authOptions *docker.AuthConf
 
 // pullImageImpl is the implementation of pulling an image. The results are
 // returned via the passed future
-func (d *dockerCoordinator) pullImageImpl(image string, authOptions *docker.AuthConfiguration, future *pullFuture) {
+func (d *dockerCoordinator) pullImageImpl(image string, authOptions *docker.AuthConfiguration, pullTimeout time.Duration, future *pullFuture) {
+	defer d.clearPullLogger(image)
 	// Parse the repo and tag
 	repo, tag := docker.ParseRepositoryTag(image)
 	if tag == "" {
 		tag = "latest"
 	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	if pullTimeout > 0 {
+		ctx, cancel = context.WithDeadline(context.Background(), time.Now().Add(pullTimeout))
```

Comment:
I tricked you into introducing a bug. This creates a new context from Background(), not `ctx`, so we're back to leaking a context again :grimacing: 

I think just change `Background() -> ctx` here

Reply:
Doh!